nautobot/docs/development/core/local-k8s.md

```diff
@@ -0,0 +1,202 @@
+
+Have a question, bug, or feature request? Let us know! https://kind.sigs.k8s.io/#community 🙂
+```
+
+To export the `kubeconfig` file with an internal cluster IP address for access from connected Nautobot containers, run the following command:
+
+```bash
+kind get kubeconfig \
+    --internal \
+    --name=nautobot \
+    > development/kind-kube-config
+```
+
+## Connecting Nautobot Containers
+
+This chapter will help you connect running Nautobot containers to the Kind network.
+
+!!! Note
+    These steps needs to be done every time you start the Nautobot containers.
+
+Verify Nautobot containers are running:
+
+```bash
+# Start Nautobot containers if they are not running
+invoke start
+
+# Verify the containers are running
+docker ps
+```
+
+!!! Info
+    In this example, we are connecting the `nautobot` and `celery_worker` containers to the Kind network to allow access to the Kind cluster from these containers during development. Connecting the `celery_worker` container is necessary if you want to run Nautobot Jobs that access the Kind Kubernetes cluster.
+
+You should see `nautobot-nautobot-1` and `nautobot-celery_worker-1` containers running and in a healthy state:
+
+```text
+CONTAINER ID   IMAGE                           COMMAND                  CREATED          STATUS          PORTS                    NAMES
+...
+1f191e104357   local/nautobot-dev:local-py3.11 "watchmedo auto-rest…"   2 minutes ago    Up 37 seconds (healthy)   0.0.0.0:8081->8080/tcp, [::]:8081->8080/tcp nautobot-celery_worker-1
+d396ed398a4c   local/nautobot-dev:local-py3.11 "/docker-entrypoint.…"   2 minutes ago    Up 2 minutes (healthy)    0.0.0.0:8080->8080/tcp, :::8080->8080/tcp nautobot-nautobot-1
+...
+```
+
+Verify the Kind Docker network name is `kind`:
+
+```bash
+docker network ls
+```
+
+You should see all networks, including the `kind` network.
+
+```text
+NETWORK ID     NAME    DRIVER    SCOPE
+...
+6de64b53be75   kind    bridge    local
+...
+```
+
+Connect running Nautobot containers to the Kind network:
+
+```bash
+docker network connect kind nautobot-nautobot-1
+docker network connect kind nautobot-celery_worker-1
+```
+
+Now the selected Nautobot containers are connected to the Kind network and can access the Kind Kubernetes cluster.
+
+## Using `kubectl`
+
+To access the Kind Kubernetes cluster from the Nautobot container, you can use `kubectl` CLI tool.
+
+!!! Note
+    If you just want to use Python scripts to interact with the cluster, you can skip this chapter. However, it's recommended to install `kubectl` as it is a powerful tool for interacting with the Kubernetes cluster for debugging and testing.
+
+To verify the connection from Nautobot container to the Kind cluster:
+
+```bash
+# Open shell in the running Nautobot container
+invoke cli
+
+# Install kubectl
+apt update
+apt install kubernetes-client
+
+# Test the connection
+export KUBECONFIG=/source/development/kind-kube-config
+kubectl cluster-info
+```
```

Comment:
Might we want to consider including `kubectl` directly in the Nautobot `dev` docker image (i.e. as an addition to the Dockerfile)? Would appreciate your opinion here.

Reply:
`kubectl` is safe to add to the dev image.

Reply:
#6307.